### DISCUSSION TITLE
Getting Invalid Credentials Error when connecting to Office365

### DISCUSSION THREAD

**Discussion by nydasco** [ecederstrand/exchangelib#1195] [Q&A]
Hi,

Not sure if someone is able to help or not.

I'm running a very basic start point, on an email address within our work domain. There is no multi-factor enabled, and I can log in with the account through the webpage. My account is accessible via outlook.office.com.

The error I'm getting is suggesting that my username/password are incorrect per [https://outlook.office365.com/EWS/Exchange.asmx](https://outlook.office365.com/EWS/Exchange.asmx). If I go to this url, I get a basic username/password popup, and my credentials indeed don't work. Is this address looking for different credentials to my basic email address and password? Do I need elevated permissions to use this Python package?

Any support would be greatly appreciated. I can see huge value for us in this package if I can get it working with our companies Azure environment.

**Comment by ecederstrand** [comment]
I believe O365 have disabled basic authentication. You need to set up OAuth instead. There's some documentation at https://ecederstrand.github.io/exchangelib/#oauth-on-office-365

**Reply by nydasco** [reply]
Ah, gotcha. My account doesn't have MFA, but it doesn't mean this isn't the reason. Not sure I'll be able to get an Azure Admin to give this full access to all mailboxes though. That seems like a big ask. I may need to look for an alternative solution.

**Reply by ecederstrand** [reply]
It should also be possible to create OAuth permissions with less than full_access_as_app. There are some hints in https://github.com/ecederstrand/exchangelib/issues/1119 but I haven't gotten around to testing and documenting.
